models/public/mozilla-deepspeech-0.8.2/mozilla-deepspeech-0.8.2.md

```diff
@@ -0,0 +1,138 @@
+
+## Accuracy
+
+| Metric                       | Value      | Parameters                                                       |
+| ---------------------------- | ---------- | ---------------------------------------------------------------- |
+| WER @ LibriSpeech test-clean | 8.39%      | with LM, beam_width = 32, Python CTC decoder, accuracy checker   |
+| WER @ LibriSpeech test-clean | **6.13%**  | with LM, **beam_width = 500**, C++ CTC decoder, accuracy checker |
+| WER @ LibriSpeech test-clean | **6.15%**  | with LM, **beam_width = 500**, C++ CTC decoder, demo             |
+
+*NB*: beam_width=32 is a low value for a CTC decoder, and was used to achieve reasonable evaluation time with Python CTC decoder in Accuracy Checker.
+Increasing beam_width improves WER metric and slows down decoding.  Speech recognition demo has a faster C++ CTC decoder module.
+
+## Performance
+
+## Input
+
+### Original Model
+
+ * **Audio MFCC coefficients**, name: `input_node` , shape: [1x16x19x26], format: [BxNxTxC], where:
+
+    - B - batch size, fixed to 1
+    - N - `input_lengths`, number of audio frames in this section of audio
+    - T - context frames: along with the current frame, the network expects 9 preceding frames and 9 succeeding frames. The absent context frames are filled with zeros.
+    - C - 26 MFCC coefficients per each frame
+
+   See [`accuracy-check.yml`](accuracy-check.yml) for all audio preprocessing and feature extraction parameters.
+
+ * Number of audio frames, INT32 value, name: `input_lengths`, shape [1].
+
+ * LSTM in-state (*c*) and input (*h*, a.k.a hidden state) vectors. Names: `previous_state_c` and `previous_state_h`, shapes: [1x2048], format: [BxC].
+
+    When splitting a long audio into chunks, these inputs must be fed with the corresponding outputs from the previous chunk.
+    Chunk processing order must be from early to late audio positions.
+
+
+### Converted Model
+
+ * **Audio MFCC coefficients**. Image, name: `input_node` , shape: [1x16x19x26], format: [BxNxTxC], where:
+
+    - B - batch size, fixed to 1
+    - N - number of audio frames in this section of audio, fixed to 16
+    - T - context frames: along with the current frame, the network expects 9 preceding frames and 9 succeeding frames. The absent context frames are filled with zeros.
+    - C - 26 MFCC coefficients in each frame
+
+    See [`accuracy-check.yml`](accuracy-check.yml) for all audio preprocessing and feature extraction parameters.
+
+ * LSTM in-state and input vectors. Names: `previous_state_c` and `previous_state_h`, shapes: [1x2048], format: [BxC].
+When splitting a long audio into chunks, these inputs must be fed with the corresponding outputs from the previous chunk.
+Chunk processing order must be from early to late audio positions.
+
+## Output
+
+### Original Model
+
+ * Per-frame probabilities (after softmax) for every symbol in the alphabet, name: `logits`, shape: [16x1x29], format: [NxBxC]
```

Comment:
Is `[NxBxC]` correct shape or it should be `[BxNxC]`?

Reply:
Here 16 is sequence length (for LSTM), 1 is batch size, 29 in the number of channels (or features). What would be the correct shape in this case?

Reply:
OK with leaving [NxBxC]?  I'm not completely sure which one is correct (because I do not fully understand what it means), but [NxBxC] looks more correct to me.